src/renderers/shared/fiber/ReactFiberScheduler.js

```diff
@@ -445,74 +446,100 @@ module.exports = function<T, P, I, TI, C>(config : HostConfig<T, P, I, TI, C>) {
     }
 
     isPerformingTaskWork = true;
-    nextUnitOfWork = findNextUnitOfWork();
-    while (nextUnitOfWork &&
-           nextPriorityLevel === TaskPriority) {
-      nextUnitOfWork =
-        performUnitOfWork(nextUnitOfWork);
+    try {
+      nextUnitOfWork = findNextUnitOfWork();
+      while (nextUnitOfWork &&
+             nextPriorityLevel === TaskPriority) {
+        nextUnitOfWork =
+          performUnitOfWork(nextUnitOfWork);
 
-      if (!nextUnitOfWork) {
-        nextUnitOfWork = findNextUnitOfWork();
+        if (!nextUnitOfWork) {
+          nextUnitOfWork = findNextUnitOfWork();
+        }
       }
+      if (nextUnitOfWork) {
+        scheduleCallbackAtPriority(nextPriorityLevel);
+      }
+    } finally {
+      isPerformingTaskWork = false;
```

Comment:
I know but we're completely moving this code around every week. I'm worried microoptimizations just make it hard to follow right now. Let's do a pass as soon as it stabilizes a little bit?